src/sentry/api/endpoints/organization_details.py

```diff
@@ -380,6 +414,25 @@ def save(self):
             org.name = data["name"]
         if "slug" in data:
             org.slug = data["slug"]
+        if (
+            features.has("organizations:api-auth-provider", org)
+            and "providerKey" in data
+            and "providerConfig" in data
+        ):
+            provider_key = data["providerKey"]
+            provider_config = data["providerConfig"]
+            with transaction.atomic():
+                auth_provider = AuthProvider.objects.update_or_create(
+                    organization_id=org.id,
+                    defaults={"provider": provider_key},
+                )[0]
+                provider = auth_provider.get_provider()
+                try:
+                    config = provider.build_config(provider_config)
+                    config["sentry-source"] = "api-organization-details"
+                except KeyError:
+                    raise KeyError(f"Invalid providerConfig for authprovider {provider_key}")
```

Comment:
Would it be better to return the original KeyError, rather than rebuilding one with less information? Mostly because there's many things that can go wrong and it might be easier to just let the user debug it.

Otherwise, is it possible to find the specific keys that are error-ing out and include it in the message to user?

Reply:
Changed error message to `Invalid providerConfig for authprovider google. Invalid Key: 'domain'`